dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -405,4 +414,44 @@ public static long countFailures(List<ArtifactResults> table, String columnName)
         .filter(row -> row.getResult(columnName) == null || row.getFailureCount(columnName) > 0)
         .count();
   }
+
+  /**
+   * Returns mapping from jar file to simplified description of its {@link DependencyPath}s. A
+   * simplified description of {@link DependencyPath}s explains a common pattern ({@code
+   * groupId:artifactId}) in the path elements, helping to avoid listing more than 5 repetitive
+   * items in BOM dashboard.
+   */
+  private static ImmutableMap<String, String> simplifyDependencyPathMessage(
+      Multimap<Path, DependencyPath> jarToDependencyPaths) {
+    // Freemarker is not good at handling non-string key
+    // https://freemarker.apache.org/docs/app_faq.html#faq_nonstring_keys
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+
+    for (Path jar : jarToDependencyPaths.keySet()) {
+      List<Set<String>> commonVersionLessCoordinates = Lists.newArrayList();
+      Collection<DependencyPath> dependencyPaths = jarToDependencyPaths.get(jar);
+      for (DependencyPath dependencyPath : dependencyPaths) {
+        // Set of "groupId:artifactId"
+        Set<String> versionLessCoordinates =
+            dependencyPath.getPath().stream().map(Artifacts::makeKey).collect(Collectors.toSet());
+        commonVersionLessCoordinates.add(versionLessCoordinates);
+      }
+      Set<String> intersection = Sets.newHashSet(Iterables.getFirst(commonVersionLessCoordinates,
+          Sets.newHashSet()));
+      for (Set<String> versionLessCoordinates : commonVersionLessCoordinates) {
+        intersection.retainAll(versionLessCoordinates);
+      }
+
+      if (dependencyPaths.size() > 5 && intersection.size() > 1) {
+        StringBuilder messageBuilder = new StringBuilder();
```

Comment:
I need to think about rewriting the message to emphasize the true problem rather than the first artifact it's found in

Reply:
Updated message as below. Let me know if you come up with something better.

> There are 994 paths to the artifact. All of them have the same artifacts: log4j:log4j, commons-logging:commons-logging, com.google.http-client:google-http-client. Example path: com.google.cloud:google-cloud-core:1.59.0 / com.google.http-client:google-http-client:1.27.0 / com.google.android:android:1.5_r4 / commons-logging:commons-logging:1.1.1 / log4j:log4j:1.2.12

Reply:
You're losing the order here. I think we want something like 

com.google.http-client:google-http-client > commons-logging:commons-logging > log4j:log4j 

Also, don't we want versions here? Or maybe not. 

And what happened to android in this example? So I think something like

log4j:log4j:1.2.12 is referenced from com.google.http-client:google-http-client 
via the path
com.google.http-client:google-http-client:1.27.0 > com.google.android:android:1.5_r4 > commons-logging:commons-logging:1.1.1 > log4j:log4j:1.2.12

Reply:
Updated to use `LinkedHashSet` to keep order of common elements.

# Version
I omit version because the paths for this log4j case are more diverse than I thought when we discussed. There are multiple versions of `com.google.http-client:google-http-client`:

![image](https://user-images.githubusercontent.com/28604/52652523-3b972000-2ebc-11e9-951a-45384fafe403.png)

# Android

There is a path that does not have android (the screenshot above)